src/app/constructor/map/map-constructor.component.ts

```diff
@@ -1,166 +1,167 @@
-            fileItem.withCredentials = false;
-            this.reader.readAsDataURL(fileItem._file);
-        });
-
-        this.uploader.onCompleteAll = () => this.onCreateSuccess();
-
-        this.reader.onload = this.ngZone.run(() => (event: any) => {
-            this.picture = event.target.result;
-        });
-    }
-
-    private initializeEditedView(mapView: MapViewInfoDto): void {
-        this.name = mapView.name;
-        this.description = mapView.description;
-        this.picture = this.mapViewService.resolvePictureUrl(mapView);
-        this.edittedDevices = mapView.sensors.map(sensor => {
-            sensor.sensor.posX = sensor.position.x;
-            sensor.sensor.posY = sensor.position.y;
-            return sensor.sensor;
-        });
-    }
-
-    private onCreateSuccess(): void {
-        this.router.navigate(['..']);
-    }
-
-    sensorIsUnique(sensor: Device): boolean {
-        return this.edittedDevices
-            .some(s => s._id === sensor._id);
-    }
-
-    public onAddSensor(sensor: Device): void {
-        if (this.picture) {
-            if (!this.sensorIsUnique(sensor)) {
-                this.edittedDevices.push(sensor);
-            }
-        }
-    }
-
-    public onRemoveSensor(sensor: Device): void {
-        if (this.picture) {
-            this.edittedDevices = this.edittedDevices.filter(s => s._id !== sensor._id);
-        }
-    }
-
-    public fileOverBase(e: any): void {
-        this.hasBaseDropZoneOver = e;
-    }
-
-    public onUploadClick() {
-        this.uploader.clearQueue();
-        const event = new MouseEvent('click', { bubbles: true });
-        this.renderer.invokeElementMethod(this.fileInput.nativeElement, 'dispatchEvent', [event]);
-    }
-
-    public onSubmit(): void {
-        if (!this.isMapViewCanBeSaved()) {
-            this.toastr.error('Please fill mandatory fields: "Name", "Description" and "Add Picture"');
-            return;
-        }
-        const mapViewInfoCreateDto: MapViewInfoCreateDto = {
-            name: this.name,
-            description: this.description,
-            active: this.isActive,
-            default: this.default === 'Map',
-            sensors: this.edittedDevices.map(({ _id, posX, posY }) => {
-                return {
-                    sensor: _id,
-                    position: {
-                        x: posX,
-                        y: posY
-                    }
-                };
-            })
-        };
-
-        this.mapViewService.create(mapViewInfoCreateDto)
-            .subscribe((mapViewInfoDto) => {
-                this.uploader.setOptions({
-                    url: this.mapViewService.resolvePictureUploadUrl(mapViewInfoDto)
-                });
-                this.uploader.uploadAll();
-            });
-    }
-
-    private isMapViewCanBeSaved(): boolean {
-        return this.name
-            && this.description
-            && this.picture;
-    }
-}
+// TODO: Refactor with subviews
```

Comment:
You can just comment using of the component in templates and declarations instead of whole file.